src/grt/src/GlobalRouter.cpp

```diff
@@ -2227,6 +2227,131 @@ void GlobalRouter::readSegments(const char* file_name)
           GRT, 261, "Error reading global route segments file {}.", file_name);
     }
   }
+  for (auto& [net, segments] : routes_) {
+    if (!isConnected(net)) {
+      logger_->error(
+          GRT, 262, "Net {} has disconnected segments.", net->getName());
+    }
+    for (const Pin& pin : db_net_map_[net]->getPins()) {
+      bool pin_is_covered = false;
+      for (const GSegment& seg : segments) {
+        // Pin is horizontally covered by the segment
+        if (pin.getOnGridPosition().getX() >= std::min(seg.init_x, seg.final_x)
+            && pin.getOnGridPosition().getX()
+                   <= std::max(seg.init_x, seg.final_x)
+            // Pin is vertically covered by the segment
+            && pin.getOnGridPosition().getY()
+                   >= std::min(seg.init_y, seg.final_y)
+            && pin.getOnGridPosition().getY()
+                   <= std::max(seg.init_y, seg.final_y)
+            // Pin and segment share a layer
+            && pin.getConnectionLayer()
+                   >= std::min(seg.init_layer, seg.final_layer)
+            && pin.getConnectionLayer()
+                   <= std::max(seg.init_layer, seg.final_layer)) {
+          pin_is_covered = true;
+          break;
+        }
+      }
+      if (!pin_is_covered) {
+        logger_->error(GRT,
+                       263,
+                       "Pin {} is not covered by net {}",
+                       pin.getName(),
+                       net->getName());
+      }
+    }
+  }
+}
+
+// Checks if segment is a line, i.e. only varies in one dimension
+// (vias are lines)
+bool GlobalRouter::segmentIsLine(const GSegment& segment)
+{
+  int dimensionality = (segment.init_x != segment.final_x)
+                       + (segment.init_y != segment.final_y)
+                       + (segment.init_layer != segment.final_layer);
+  return (dimensionality == 1);
+}
+
+// Implements Union-Find algorithm to determine connectivity
+bool GlobalRouter::isConnected(odb::dbNet* net)
+{
+  std::vector<int> parent(routes_[net].size()), rank(routes_[net].size(), 0);
+  for (int i = 0; i < routes_[net].size(); i++) {
+    parent[i] = i;
+  }
+  int initialized_groups = 1;
+
+  if (!segmentIsLine(routes_[net][0])) {
+    logger_->error(
+        GRT,
+        264,
+        "Segment {} of net {} is not a horizontal/vertical line or via",
+        1,
+        net->getName());
+  }
+
+  std::function<int(int)> find = [&](int x) -> int {
+    if (parent[x] != x) {
+      parent[x] = find(parent[x]);
+    }
+    return parent[x];
+  };
+
+  std::function<void(int, int)> uniteGroups = [&](int u, int v) {
+    int rootU = find(u), rootV = find(v);
```

Comment:
Use the same convention for variable names. Instead of rootU/rootV, use root_u/root_v.